pkg/controller/replication/conversion.go

```diff
@@ -315,23 +275,23 @@ func convertCall(fn func(*v1.ReplicationController) (*v1.ReplicationController,
 	return convertRCtoRS(result, nil)
 }
 
-func convertRCtoRS(rc *v1.ReplicationController, out *extensionsv1beta1.ReplicaSet) (*extensionsv1beta1.ReplicaSet, error) {
+func convertRCtoRS(rc *v1.ReplicationController, out *apps.ReplicaSet) (*apps.ReplicaSet, error) {
 	var rsInternal extensions.ReplicaSet
 	if err := apiv1.Convert_v1_ReplicationController_to_extensions_ReplicaSet(rc, &rsInternal, nil); err != nil {
 		return nil, fmt.Errorf("can't convert ReplicationController %v/%v to ReplicaSet: %v", rc.Namespace, rc.Name, err)
 	}
 	if out == nil {
-		out = new(extensionsv1beta1.ReplicaSet)
+		out = new(apps.ReplicaSet)
 	}
-	if err := extensionsinternalv1beta1.Convert_extensions_ReplicaSet_To_v1beta1_ReplicaSet(&rsInternal, out, nil); err != nil {
+	if err := appsconversion.Convert_extensions_ReplicaSet_To_v1_ReplicaSet(&rsInternal, out, nil); err != nil {
```

Comment:
Whether we can try convert RC to RS don't through extension? :-)

Reply:
@fisherxu Do you have a concern about the correctness of the code? Or is this just a comment on the fact that this looks weird?

For context on why it looks weird: The internal (unversioned) [ReplicaSet struct](https://github.com/kubernetes/kubernetes/blob/bcf517e37251cf41ecb7e1fd1f5fd23a7458fcf6/pkg/apis/extensions/types.go#L679-L692) is still in `extensions`. So to convert between `apps/v1` ReplicaSet and `core/v1` ReplicationContorller, we go via the unversioned internal `extensions` ReplicaSet.

Reply:
In fact, I have tried to bump deployment/replicaset to apps/v1 last week, but haven't dealt with the rollback problem yet :-( and so not send a PR. Then observed your PR, so great~
To this conversion, I also through  extensions in my local env. I was thinking if we can move internal RS/Deployment to apps in a follow-up PR?

Reply:
@enisoc If we need move internal RS/Deployment to apps, I can help do it in a follow-up PR :)

Reply:
@fisherxu Sorry about overlapping with your work. Did you post somewhere that you were working on that? FYI I started Deployment in #61419 but that's currently blocked by #61433.

I agree we should move the internal struct to `apps` in a separate PR. If you plan to work on that, please leave a comment on #55714 so we don't overlap. Thanks!

Reply:
@enisoc Has sent a PR #61755 to move the internal struct to apps, and working on the failed CI :)